### PR TITLE
refactor: unifies models to share fields validation rules

### PR DESCRIPTION
**Pull request by byashimov**
- Moves secret fields to a dedicated model
- Moves `DiskSpace` to `ServiceCommonSpec`
- Renders secret keys in admonition

### CODE REVIEW

**Comment by byashimov** [comment]
This is how the secret admonition looks like:
image

**Comment by byashimov** [comment]
[tests](https://github.com/aiven/aiven-operator/actions/runs/8661404767)

File: generators/docs/generator.go [hunk 1]
```diff
@@ -220,8 +220,13 @@ func (s *schemaType) IsNested() bool {
...
 }
 
-// reInlineCode finds all code-like text
-var reInlineCode = regexp.MustCompile(`'(\S+)'`)
+var (
+	// reInlineCode to wrap inline code into backticks
+	reInlineCode = regexp.MustCompile(`'(\S+)'`)
+
+	// reAdmonitions to render markdown admonitions
+	reAdmonitions = regexp.MustCompile(`(?i)((?:note|tip|info|warning|danger|bug|success|example) "[^"]*"):\s*(.+)`)
```
Comment (Serpentiel) [inline generators/docs/generator.go:228]: is it possible to apply some markdown library instead, or will it be too much?

Comment (byashimov) [reply generators/docs/generator.go:228]: It's not markdown. It is a plain text that I try to turn into markdown.

**Review by Serpentiel — APPROVED** [review]
lgtm

**Pull request merged**